modules/oe_whitelabel_contact_forms/tests/src/Kernel/ContactFormRenderTest.php

```diff
@@ -0,0 +1,130 @@
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests that our contact forms renders with right markup.
+ */
+class ContactFormRenderTest extends KernelTestBase {
+
+  use SparqlConnectionTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'oe_bootstrap_theme_helper',
+    'ui_patterns',
+    'ui_patterns_library',
+    'path',
+    'path_alias',
+    'options',
+    'user',
+    'system',
+    'telephone',
+    'contact',
+    'contact_storage',
+    'rdf_skos',
+    'oe_contact_forms',
+    'oe_corporate_countries',
+    'sparql_entity_storage',
+    'oe_whitelabel_contact_forms',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    \Drupal::service('theme_installer')->install(['oe_whitelabel']);
+    $this->installEntitySchema('path_alias');
+    $this->installEntitySchema('contact_message');
+    $this->installEntitySchema('user');
+    $this->installSchema('system', 'sequences');
+
+    \Drupal::configFactory()
+      ->getEditable('system.theme')
+      ->set('default', 'oe_whitelabel')
+      ->save();
+    $this->container->set('theme.registry', NULL);
+    $this->setUpSparql();
+  }
+
+  /**
+   * Tests that corporate contact form is rendered with the correct ECL markup.
+   */
+  public function testContactForm(): void {
+    $contact_form = ContactForm::create(['id' => 'oe_contact_form']);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'header', 'this is a test header');
+    $privacy_url = 'http://example.net';
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'privacy_policy', $privacy_url);
+    $optional_selected = ['oe_telephone' => 'oe_telephone'];
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'optional_fields', $optional_selected);
+    $topics = [
+      [
+        'topic_name' => 'Topic name',
+        'topic_email_address' => 'topic@emailaddress.com',
+      ],
+    ];
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'topics', $topics);
+    $contact_form->save();
+
+    /** @var \Drupal\contact\MessageInterface $contact_message */
+    $message = $this->container->get('entity_type.manager')->getStorage('contact_message')->create([
+      'contact_form' => $contact_form->id(),
+      'name' => 'sender_name',
+      'mail' => 'test@example.com',
+      'subject' => 'subject',
+      'message' => 'welcome_message',
+      'oe_telephone' => '0123456',
+      'oe_topic' => 'Topic name',
+    ]);
+
+    /** @var \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager */
+    $entity_type_manager = $this->container->get('entity_type.manager');
+    $form = $this->container->get('entity.form_builder')->getForm($message, 'corporate_default');
+    $crawler = new Crawler($this->render($form));
+
+    // Assert classes and contact form.
+    $actual = $crawler->filter('hr.mt-5.mb-5');
```

Comment:
after we get the form as mentioned below we assert only the elements we alter, but in the context of the form
```suggestion
    $wrapper = $form->filter('fieldset');
    $hr = $wrapper->filter('hr.mt-5.mb-5');
    next assert the 2 fields are present..
```

Reply:
I appended it into the fieldset prefix so it will fail in this way:
![image](https://user-images.githubusercontent.com/6443180/154735016-08331083-1629-4c0e-a526-f56c07a813ec.png)